source/AlgebraicMethods/PolyReader.cpp

```diff
@@ -69,19 +69,17 @@ XTerm *PolyReader::_ReadXTerm(char *stream, int s, int e) {
     e1 = _GotoCloseBracket(stream, s1 + 1, e);
     _Assert(e1 >= 0, "Right bracket missing!");
 
-    UPolynomialFraction *uf = _ReadUFraction(stream, s1, e1);
+    std::shared_ptr<UPolynomialFraction> uf = _ReadUFraction(stream, s1, e1);
     xt->SetUFraction(uf);
-    uf->Dispose();
   } else {
     // type 2
 
     // read real coeff
     REAL cf;
     sscanf(&stream[e1 + 1], "%lf", &cf);
 
-    UPolynomialFraction *uf = new UPolynomialFraction(cf);
+    auto uf = std::make_shared<UPolynomialFraction>(cf);

```

Comment:
In changes like this, I’ve used `auto` where the type is clear from being explicit on the RHS of the assignment. Please let me know if you object to this style.

Reply:
auto is not used elsewhere in the project, and I think there is no need to use/introduce it, i.e. I certainly would not introduce it systematically in the code now.

Reply:
On the other hand, the code for algebraic provers always had its specifics (and now it uses shared pointers), so it is ok to have "auto" (only) there.